pkg/controllers/cluster/import.go

```diff
@@ -259,7 +259,15 @@ func (i *importHandler) restConfigFromKubeConfig(data []byte) (*rest.Config, err
 		return nil, err
 	}
 
-	if raw.Contexts[raw.CurrentContext] != nil {
+	if raw.Contexts[clusterName] != nil {
+		cluster := raw.Contexts[clusterName].Cluster
+		if raw.Clusters[cluster] != nil {
+			_, err := http.Get(raw.Clusters[cluster].Server)
+			if err == nil {
+				raw.Clusters[cluster].CertificateAuthorityData = nil
+			}
+		}
+	} else if raw.Contexts[raw.CurrentContext] != nil {
```

Comment:
Does Fleet maintain backwards compatible behavior even with the context check moved after the new cluster name check?

Reply:
I think this could be backwards compatible if (1) the context entry for a cluster is named differently than the cluster entry _and_ (2) the cluster name provided by the user is also used as the name for a context in the config that points to a _different_ cluster.

Reply:
Great point @genevieve. Perhaps, it might be a good idea to add checks here for those two scenarios.

Reply:
Ah that makes sense. Regardless, I think ensuring backwards compatibility is essential for this PR since Fleet auto-updates with Rancher currently.

Reply:
I can swap the if checks so that it continues to default to selecting by the current context?

Reply:
That would be a good idea, but then we might never use `cluster.Name` as context override. We may want to keep the same if branch order, but add a check.

```go
cluster := raw.Contexts[clusterName].Cluster
if raw.Clusters[cluster] != nil {}
```

We may want to default to the previous behavior if this is `nil`. 

CC: @kinarashah @StrongMonkey

Reply:
I updated the PR to only use the newer codepath if `raw.Contexts[clusterName] != nil` otherwise it will fallback to the older behaviour for backwards compatibility
```go
// restConfigFromKubeConfig checks kubeconfig data and tries to connect to server. If server is behind public CA, remove CertificateAuthorityData in kubeconfig file.
func (i *importHandler) restConfigFromKubeConfig(clusterName string, data []byte) (*rest.Config, error) {
	clientConfig, err := clientcmd.NewClientConfigFromBytes(data)
	if err != nil {
		return nil, err
	}

	raw, err := clientConfig.RawConfig()
	if err != nil {
		return nil, err
	}

	if raw.Contexts[clusterName] != nil {
		cluster := raw.Contexts[clusterName].Cluster
		if raw.Clusters[cluster] != nil {
			_, err := http.Get(raw.Clusters[cluster].Server)
			if err == nil {
				raw.Clusters[cluster].CertificateAuthorityData = nil
			}
		}

		return clientcmd.NewDefaultClientConfig(raw, &clientcmd.ConfigOverrides{CurrentContext: clusterName}).ClientConfig()
	}

	if raw.Contexts[raw.CurrentContext] != nil {
		cluster := raw.Contexts[raw.CurrentContext].Cluster
		if raw.Clusters[cluster] != nil {
			_, err := http.Get(raw.Clusters[cluster].Server)
			if err == nil {
				raw.Clusters[cluster].CertificateAuthorityData = nil
			}
		}
	}

	return clientcmd.NewDefaultClientConfig(raw, &clientcmd.ConfigOverrides{}).ClientConfig()
}
```